jenkins-build-pack-api/Jenkinsfile

```diff
@@ -107,207 +108,128 @@ node() {
-				}
-			}//stage to prod ends here
-
+		def env_key
+		if(branch == "master") {
+			env_key = "PROD"
+		}else{
+			env_key = "DEV"
+		}
+		stage("Deployment to ${env_key} environment") {
+			sendEvent(service, branch, runtime, env_key, configLoader.AWS.REGION, domain, configLoader.AWS.ROLEID, Event_Name.DEPLOY_TO_AWS, Event_Status.STARTED, "starting")
+			echo "starts Deployment to ${env_key} Env"
 
-		} else {
-			stage('Deployment to Dev Env') {
-				sendEvent(service, branch, runtime, "DEVELOPMENT", configLoader.AWS.REGION, domain, configLoader.AWS.ROLEID, Event_Name.DEPLOY_TO_AWS, Event_Status.STARTED, "starting")
-				echo "starts Deployment to Development Env"
+			// sh "cp serverless-prod.yml serverless.yml"
+			// sh "cp swagger/swagger-prod.json swagger/swagger.json"
+			sh "rm -rf .serverless"
 
-				withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
+			withCredentials([[$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']]) {
 				echo "AWS Configure ......."
 				sh "aws configure set profile.cloud-api.region ${configLoader.AWS.REGION}"
 				sh "aws configure set profile.cloud-api.aws_access_key_id $AWS_ACCESS_KEY_ID"
 				sh "aws configure set profile.cloud-api.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
 
-				loadServerlessConfig(runtime, config)
-
 				// Generate serverless yml file with domain added in function name
-				echo "Generate deployment env with domain"
-
-				writeServerlessFile(config)
+				echo "Generate deployment env with domain for ${env_key}"
 				
-				generateServerlessYml(service, current_environment, serviceName, cfStackName)
+				writeServerlessFile(config, current_environment)				
 				
 				setLambdaExecutionRole(configLoader.AWS.ROLEID);
 
 				echo "serverless deploy......."
-
-				sh "serverless deploy --stage dev -v --bucket " + configLoader.AWS.S3.DEV_BUCKET
-				createSubscriptionFilters(serviceName, current_environment, configLoader.AWS.REGION, logStreamer);
-
-				def svc_status = ''
+				def envBucketKey = "${env_key}${configLoader.JAZZ.S3_BUCKET_NAME_SUFFIX}"
+				sh "serverless deploy --stage ${current_environment} -v --bucket ${configLoader.AWS.S3[envBucketKey]}"
+								
 				if (fileExists('swagger/swagger.json')) {
 					// Generate swagger file based on environment
 					echo "Generating the swagger file for each environment"
-					
-					def apiHostName = "${configLoader.AWS.API.DEV_ID}.execute-api.${configLoader.AWS.REGION}.amazonaws.com" //Staging environment
+
+					def api_key = env_key + "_ID"
+					def apiHostName = "${configLoader.AWS.API[api_key]}.execute-api.${configLoader.AWS.REGION}.amazonaws.com" //Staging environment
```

Comment:
${configLoader.AWS.API[api_key] => This makes me think, we should make each env as a node under API, that way you can reference it as [env] instead of env_ID

Reply:
Awesome thought Surya.  The same we can apply in the jazz s3 bucket. So that we can avoid the bucket name suffix/prefix